projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -362,6 +375,19 @@ export class MultiSelectComponent<V> implements ControlValueAccessor, AfterConte
     this.propagateControlValueChange?.(value);
     this.propagateControlValueChangeOnTouch?.(value);
   }
+
+  private reorderSelectedItemsFirst(
+    filteredOptions: SelectOptionComponent<V>[],
+    selected?: V[]
+  ): SelectOptionComponent<V>[] {
+    if (isEmpty(selected)) {
+      return filteredOptions;
+    }
+    const filteredOptionsPartitions = compact(
+      partition(filteredOptions, filteredOption => selected?.includes(filteredOption.value))
+    );
+    return [...filteredOptionsPartitions[0], ...filteredOptionsPartitions[1]];
```

Comment:
Will there always be two items in this array if we use `compact`? Perhaps a unit test would make this more understandable?

Reply:
it's splitting the array between selected and not selected items